cognite/seismic/protos/v1/seismic_service_datatypes.proto

```diff
@@ -300,3 +318,39 @@ enum TraceHeaderField {
     CROSSLINE = 4;
     SHOTPOINT = 5;
 }
+
+message GeometryFilter {
+    Geometry geometry = 1;
+    InterpolationMethod interpolation_method = 2; // Only used for linear geometries
+}
+
+message Seismic2dExtent {
+    TraceHeaderField trace_key = 1; // Must be an applicable field for 2D
+    repeated LineDescriptor trace_ranges = 2; // Range of header values to use
+}
+
+message Seismic3dExtent {
+    oneof type {
+        Seismic3dRects rects = 1;
```

Comment:
This variant of the `Seismic3dExtent` would probably only be passed by callers to the service, never returned to the caller as computed extents, at least initially. Finding an optimal description of a `GridLayout` as a union of rectangles seems like a Hard Problem.